File: 1845-seat-reservation-manager/1845-seat-reservation-manager.cpp
class SeatManager {
public:
    int marker;
    priority_queue<int, vector<int>, greater<int>> availableSeats;
    SeatManager(int n) {
        marker = 1;
    }
    
    int reserve() {
        if(!availableSeats.empty()){
            int seatNumber = availableSeats.top();
            availableSeats.pop();
            return seatNumber;
        }
        int seatNumber = marker;
        marker++;
        return seatNumber;
    }
    
    void unreserve(int seatNumber) {
        availableSeats.push(seatNumber);
    }
};

/**
 * Your SeatManager object will be instantiated and called as such:
 * SeatManager* obj = new SeatManager(n);
 * int param_1 = obj->reserve();
 * obj->unreserve(seatNumber);
 */

File: 1845-seat-reservation-manager/1845-seat-reservation-manager.py
class SeatManager:

    def __init__(self, n: int):
        self.pq = [i for i in range(1, n+1)]

    def reserve(self) -> int:
        seatNumber = heapq.heappop(self.pq)
        return seatNumber

    def unreserve(self, seatNumber: int) -> None:
        heapq.heappush(self.pq, seatNumber)


# Your SeatManager object will be instantiated and called as such:
# obj = SeatManager(n)
# param_1 = obj.reserve()
# obj.unreserve(seatNumber)